src/pools/D3MAaveDaiPool.sol

```diff
@@ -0,0 +1,149 @@
+    RewardsClaimerLike       public immutable rewardsClaimer;
+    ShareTokenLike           public immutable stableDebt;
+    ShareTokenLike           public immutable variableDebt;
+    address                  public immutable interestStrategy;
+
+    address public king;     // Who gets the rewards
+
+    event Collect(address indexed king, address[] assets, uint256 amt);
+
+    constructor(address hub_, address dai_, address pool_, address _rewardsClaimer) public D3MPoolBase(hub_, dai_) {
+        pool = pool_;
+
+        // Fetch the reserve data from Aave
+        (,,,,,,, address adai_, address stableDebt_, address variableDebt_, address interestStrategy_,) = LendingPoolLike(pool_).getReserveData(dai_);
+        require(adai_ != address(0), "D3MAaveDaiPool/invalid-adai");
+        require(stableDebt_ != address(0), "D3MAaveDaiPool/invalid-stableDebt");
+        require(variableDebt_ != address(0), "D3MAaveDaiPool/invalid-variableDebt");
+        require(interestStrategy_ != address(0), "D3MAaveDaiPool/invalid-interestStrategy");
+
+        stableDebt = ShareTokenLike(stableDebt_);
+        variableDebt = ShareTokenLike(variableDebt_);
+        interestStrategy = interestStrategy_;
+        rewardsClaimer = RewardsClaimerLike(_rewardsClaimer);
+
+        ShareTokenLike(adai_).approve(pool_, type(uint256).max);
+        TokenLike(dai_).approve(pool_, type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Math ---
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "D3MAaveDaiPool/overflow");
+    }
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, RAY) / y;
+    }
+
+    // --- Admin ---
+    function file(bytes32 what, address data) public override auth {
+        require(live == 1, "D3MAaveDaiPool/no-file-not-live");
+
+        if (what == "king") king = data;
+        else super.file(what, data);
+    }
+
+    function validTarget() external view override returns (bool) {
+        (,,,,,,,,,, address strategy,) = LendingPoolLike(pool).getReserveData(address(asset));
+        return strategy == interestStrategy;
+    }
+
+    // Deposits Dai to Aave in exchange for adai which gets sent to the msg.sender
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#deposit
+    function deposit(uint256 amt) external override auth {
+        LendingPoolLike(pool).deposit(address(asset), amt, address(this), 0);
+    }
+
+    // Withdraws Dai from Aave in exchange for adai
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#withdraw
+    function withdraw(uint256 amt) external override auth {
+        LendingPoolLike(pool).withdraw(address(asset), amt, address(hub));
+    }
+
+    // --- Collect any rewards ---
+    function collect(address[] memory assets, uint256 amount) external auth returns (uint256 amt) {
```

Comment:
This doesn't need the `auth` modifier. It's permissionless on the currently deployed D3M.